src/pages/BorrowerProfile/BorrowerProfile.vue

```diff
@@ -531,21 +537,28 @@ export default {
 		}
 
 		// Async data fetch for MARS-317 and share button
-		const { data } = await this.apollo.query({
-			query: mountedQuery,
-			variables: {
-				loanId: this.loanId,
-			},
-		});
-		const loan = data?.lend?.loan;
-		this.partnerName = loan?.partnerName ?? '';
-		this.partnerCountry = loan?.partner?.countries[0]?.name ?? '';
-		this.lender = data?.my?.userAccount ?? {};
+		if (this.enabledExperiment) {
+			const { data } = await this.apollo.query({
+				query: mountedQuery,
+				variables: {
+					loanId: this.loanId,
+				},
+			});
+			const loan = data?.lend?.loan;
+			this.partnerName = loan?.partnerName ?? '';
+			this.partnerCountry = loan?.partner?.countries[0]?.name ?? '';
+			this.lender = data?.my?.userAccount ?? {};
+		}
 	},
 	computed: {
 		loanId() {
 			return Number(this.$route.params.id || 0);
 		},
+		enabledExperiment() {
+			const hasLentBefore = this.cookieStore.get(hasLentBeforeCookie);
+			const hasDepositBefore = this.cookieStore.get(hasDepositBeforeCookie);
+			return hasLentBefore !== 'true' && hasDepositBefore !== 'true' && !this.hasEverLoggedIn;
```

Comment:
So we're only running this for the first login of a user? At what point will `hasEverLoggedIn` change from returning `true` to returning `false`? For example, will it be `true` as soon as a user is created and logged in, or is it `true` on the second login of a new user only?

Reply:
Generally, so long as cookies haven't been recently cleared, hasEverLoggedIn will return true if it can find a current login session, remnants of a login session or the kvu cookie. ref. https://github.com/kiva/ui/blob/main/src/api/localResolvers/my.js#L14

It's not "absolute" but it's fine for the purposes of experimentation. We "could" do some additional filtering in the data analysis stage to exclude existing users who may have gotten shown the exp.

Reply:
In that case, will `this.hasEverLoggedIn` only be `false` for non-logged-in users who have never logged in via their current browser? Assuming that cookie is set to `true` as soon as a user logs in.

Reply:
That's right. Hope that's your goal @christian14b !!!

Reply:
Thank you both for comments! Yeah @mcstover, that's the expected behavior!